### PR TITLE
#163894898 Refactors tests and validations based on feedback given

### PR DESCRIPTION
**Pull request by murugilinet**
#### What does this PR do?
* adds more validations on endpoints to ensure the required field inputs are correct and valid
#### Description of task to de completed?
* Add more validations to ensure the same party cannot be created twice
* Ensure the office-types can only be state,federal,legislature or local-community
#### How should this be manually tested?
* After cloning the repo,cd into it and run python run.py and pytest
* Test all endpoints on postman 
#### What are the Relevant Pivotal Tracker stories?
* [#163894898](https://www.pivotaltracker.com/story/show/163894898)

### CODE REVIEW

File: app/api/v1/model/models.py [hunk 1]
```diff
@@ -26,17 +26,38 @@ def remove(self, id):
         if item:
             return self.items.remove(item)
     
-    def update(self, id):
+    def update(self, id,data):
```
Comment (pabishai) [inline app/api/v1/model/models.py:29]: Follow pep8 standards. Install pylint to help with that

File: app/api/v1/view/office_views.py [hunk 2]
```diff
@@ -41,18 +41,24 @@ def post(self):
                 }
 
         response = make_response(jsonify({
-            'Message': 'Input is required',
+            'Message': 'Correct input required',
         }),400) 
-        if self.dt.valid(data['name']) == False: 
+        if self.dt.valid_type(data['name']) == False:
```
Comment (pabishai) [inline app/api/v1/view/office_views.py:46]: separate the validators to clean up the code better

**Review by pabishai — COMMENTED** [review]
Please consider my comments

**Comment by murugilinet** [comment]
https://us04web.zoom.us/j/460008061

**Pull request merged**